docs/packages/xstate-react/index.md

```diff
@@ -208,6 +208,29 @@ const App = ({ service }) => {
 };
 ```
 
+::: warn
+The default comparison is a strict reference comparison (`===`). If your selector returns non-primitive values, such as objects or arrays, you should keep this in mind and either return the same reference, or provide a shallow or deep comparator.
+
+One option is to use the popular [`fast-deep-equal`](https://www.npmjs.com/package/fast-deep-equal) library:
+
+```js
+import isDeepEqual from 'fast-deep-equal';
```

Comment:
I think it would be better to recommend one of the libraries for `shallowEqual`. Deep equality check can be expensive sometimes so I'm worried about people using this example from the docs too blindly.

Reply:
Good call; will replace this in the morning 😴